src/github/github.service.ts

```diff
@@ -0,0 +1,167 @@
+import {
+  BadRequestException,
+  HttpException,
+  HttpStatus,
+  Injectable,
+  Logger,
+  NotFoundException,
+} from '@nestjs/common';
+import { Octokit } from '@octokit/rest';
+import { EntityManager, In, Repository } from 'typeorm';
+import { InjectEntityManager, InjectRepository } from '@nestjs/typeorm';
+import { User } from './entities/user.entity';
+import { ApiMessages, TokenUsageTags } from 'src/common/constants';
+import { TRequest } from 'src/common/types/user.types';
+import { Env } from 'src/env';
+import * as fs from 'fs';
+import * as jwt from 'jsonwebtoken';
+import * as path from 'path';
+import axios from 'axios';
+import { Repo } from 'src/webhook/entities/repository.entity';
+import { UpdatePracticesDto } from './dto/update.practices.dto';
+import { OpenAiService } from 'src/open-ai/open-ai.service';
+import { calculateOffset } from 'src/utils/app.utils';
+import { PullRequest } from 'src/webhook/entities/pull-request.entity';
+import { UserInstallation } from './entities/user-installation.entity';
+import { GithubInstallationService } from './github.installation.service';
+
+@Injectable()
+export class GithubService {
+  private readonly logger = new Logger(GithubService.name);
+  constructor(
+    @InjectRepository(User)
+    private userRepository: Repository<User>,
+    @InjectRepository(UserInstallation)
+    private userInstallationRepository: Repository<UserInstallation>,
+    @InjectEntityManager()
+    private entityManager: EntityManager,
+    private readonly openAiService: OpenAiService,
+    private readonly githubInstallationService: GithubInstallationService,
+  ) {}
```

Comment:
PullProofAI

The constructor of `GithubService` is doing a lot of dependency injection directly. Consider refactoring this to follow the Dependency Injection principle, where dependencies are injected through the constructor rather than being directly instantiated within the class.

```diff
-     private readonly openAiService: OpenAiService,
-     private readonly githubInstallationService: GithubInstallationService,
+     @Inject(OpenAiService)
+     private readonly openAiService: OpenAiService,
+     @Inject(GithubInstallationService)
+     private readonly githubInstallationService: GithubInstallationService,
```

#### Issue: Error Handling


<!-- This is an auto-generated comment by PullProofAI -->